core/src/forwarding_stage.rs

```diff
@@ -45,95 +42,178 @@ mod packet_container;
 /// this should be evaluated with new stage.
 const FORWARD_BATCH_SIZE: usize = 128;
 
-/// Addresses to use for forwarding.
+/// [`ForwardAddressGetter`] provides helper methods for retrieving forwarding
+/// addresses for both vote and non-vote transactions.
 #[derive(Clone)]
-pub struct ForwardingAddresses {
-    /// Forwarding address for TPU (non-votes)
-    pub tpu: Option<SocketAddr>,
-    /// Forwarding address for votes
-    pub tpu_vote: Option<SocketAddr>,
-}
-
-pub trait ForwardAddressGetter: Send + Sync + 'static {
```

Comment:
My understanding is that this trait was introduced only to be able to have a lightweight implementation for the tests. So removed it.